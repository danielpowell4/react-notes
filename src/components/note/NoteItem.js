import React from 'react';
import {StaticNote} from './StaticNote'
import {EditNote} from './EditNote'

export const NoteItem = (props) => {
  if (props.isEditting) {
    return ( <EditNote id={props.id}
                       content={props.content}
                       handleUpdateContent={props.handleUpdateContent}
                       handleEditSubmit={props.handleEditSubmit}/> )
  }
  return ( <StaticNote id={props.id}
                       content={props.content}
                       handleToggleEdit={props.handleToggleEdit}
                       handleRemove={props.handleRemove}/> )
}

  NoteItem.propTypes = {
    content: React.PropTypes.string.isRequired,
    handleEditSubmit: React.PropTypes.func.isRequired,
    handleRemove: React.PropTypes.func.isRequired,
    handleToggleEdit: React.PropTypes.func.isRequired,
    handleUpdateContent: React.PropTypes.func.isRequired,
    id: React.PropTypes.number.isRequired,
    isEditting: React.PropTypes.bool.isRequired
  }
