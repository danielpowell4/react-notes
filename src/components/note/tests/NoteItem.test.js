import React from 'react';
import renderer from 'react-test-renderer';
import {NoteItem} from '../NoteItem';

/* NoteItem snapshot tests */

const staticNoteItemSampleProps = { content: 'static note toBe',
                                    handleRemove: function(){},
                                    handleToggleEdit: function(){},
                                    id: 2,
                                    isEditting: false }

test('NoteItem renders a StaticNote correctly', () => {
  const tree = renderer.create(
    <NoteItem {...staticNoteItemSampleProps} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

const edittingNoteItemSampleProps = { content: 'edit note toBe',
  handleUpdateContent: function(){},
  handleEditSubmit: function(){},
  id: 3,
  isEditting: true,
}

test('NoteItem renders an EditNote correctly', () => {
  const tree = renderer.create(
    <NoteItem {...edittingNoteItemSampleProps} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
