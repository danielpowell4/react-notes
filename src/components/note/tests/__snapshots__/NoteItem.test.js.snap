exports[`test NoteItem renders a StaticNote correctly 1`] = `
<li>
  <span
    className="delete-item">
    <a
      href="#"
      onClick={[Function]}>
      X
    </a>
  </span>
  <span
    onClick={[Function]}>
    static note toBe
  </span>
</li>
`;

exports[`test NoteItem renders an EditNote correctly 1`] = `
<form
  onSubmit={[Function]}>
  <input
    onChange={[Function]}
    type="text"
    value="edit note toBe" />
</form>
`;
